index.js

```diff
@@ -66,7 +66,8 @@ let appendSourceList = function(policyObject, name, sourceList) {
     oldSourceList = oldValue;
   }
 
-  oldSourceList.push(sourceList);
+  // do not mutate existing source list to prevent leaking state
+  oldSourceList.slice().push(sourceList);
```

Comment:
This doesn't make any sense to me. You are creating a copy of the array, pushing something into the copy and then throwing the copy away (all on the same line).

Am I missing something obvious? Or is this code a no-op?

Reply:
Good catch! Demonstrates how bad our test coverage still is.

This method is awful. Will add tests covering that one and refactor it in another PR.